Date: Sun, 03 Apr 2022 11:15:44 +0300
Subject: Fix build with CGAL-5.4
Upstream: modified from upstream commits
- c32efe043a65b7fd761751c8edb56f8deb6a9ed5
- 71f2831c0484c3f35cbf44e1d1dc2c857384100b

diff --git a/src/cgalutils-tess.cc b/src/cgalutils-tess.cc
index ec1cc1e..fb709fc 100644
--- a/src/cgalutils-tess.cc
+++ b/src/cgalutils-tess.cc
@@ -6,10 +6,12 @@
 #pragma push_macro("NDEBUG")
 #undef NDEBUG
 #include <CGAL/Constrained_Delaunay_triangulation_2.h>
-#if CGAL_VERSION_NR >= CGAL_VERSION_NUMBER(4,11,0)
-  #include <CGAL/Triangulation_2_projection_traits_3.h>
+#if CGAL_VERSION_NR < CGAL_VERSION_NUMBER(5, 4, 0)
+#include <CGAL/Triangulation_2_projection_traits_3.h>
+typedef CGAL::Triangulation_2_filtered_projection_traits_3<K> Projection;
 #else
-  #include <CGAL/Triangulation_2_filtered_projection_traits_3.h>
+#include <CGAL/Projection_traits_3.h>
+typedef CGAL::Filtered_projection_traits_3<K> Projection;
 #endif
 #include <CGAL/Triangulation_face_base_with_info_2.h>
 #pragma pop_macro("NDEBUG")
@@ -19,7 +21,6 @@ struct FaceInfo {
   bool in_domain() { return nesting_level%2 == 1; }
 };
 
-typedef CGAL::Triangulation_2_filtered_projection_traits_3<K> Projection;
 typedef CGAL::Triangulation_face_base_with_info_2<FaceInfo, K> Fbb;
 typedef CGAL::Triangulation_data_structure_2<
 	CGAL::Triangulation_vertex_base_2<Projection>,
